x/btcstaking/keeper/ibc_callbacks.go

```diff
@@ -142,7 +142,16 @@ func (k Keeper) processAddBsnRewards(
 		return status.Errorf(codes.InvalidArgument, "invalid address %s: %v", transferData.Receiver, err)
 	}
 
-	return k.AddBsnRewards(ctx, receiverOnBbnAddr, callbackAddBsnRewards.BsnConsumerID, sdk.NewCoins(bsnReward), callbackAddBsnRewards.FpRatios)
+	bsnConsID := callbackAddBsnRewards.BsnConsumerID
+	if len(bsnConsID) == 0 {
+		// uses the destination because it is the port and channel registered on babylon side
```

Comment:
@SebastianElvis @gitferry please confirm that this should be the Dest part of the packet, not source